src/features/wallet/TwoKeyContractModal.tsx

```diff
@@ -0,0 +1,81 @@
+/** @jsxImportSource @emotion/react */
+import { ForwardRefRenderFunction, PropsWithChildren, forwardRef } from "react";
+import tw from "twin.macro"; // eslint-disable-line
+import Modal from "common/components/modal";
+import { ModalWrapper } from "./styled/TwoKeyContractModal.styled";
+import { Disconnected, Connected } from "./styled/Wallet.styled";
+import createDesignatedVotingContract from "./helpers/createDesignatedVotingContract";
+import { ethers } from "ethers";
+interface Props {
+  isOpen: boolean;
+  close: () => void;
+  ref: (node: HTMLElement | null) => void;
+  hotAddress: string | null;
+  votingAddress: string | null;
+  isConnected: boolean;
+  network: ethers.providers.Network | null;
+  signer: ethers.Signer | null;
+}
+
+const _TwoKeyContractModal: ForwardRefRenderFunction<
+  HTMLElement,
+  PropsWithChildren<Props>
+> = (
+  { isOpen, close, hotAddress, votingAddress, isConnected, network, signer },
+  externalRef
+) => {
+  return (
+    <Modal isOpen={isOpen} onClose={close} ref={externalRef}>
+      <ModalWrapper>
+        <h3 className="header">Two Key Voting</h3>
+        {!isConnected ? (
+          <p tw="opacity-50 mb-4 text-center">
+            Connect your wallet to start this process.
+          </p>
+        ) : (
+          <>
+            <p tw="opacity-50 mb-4 text-center">
+              {!hotAddress
+                ? `You are not currently using a two key voting system. To deploy one,
+          provide your cold key address. Click here to learn more about the two
+          key voting system.`
+                : `Your hot wallet address is: ${hotAddress}`}
```

Comment:
could we also show the cold wallet address?

Reply:
Yup!